scripts/compile_modules.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Compiles the given model to modules. The targets are expected to be a
+# comma-separated list, e.g., --targets=vmla,vulkan,dylib
+#
+# If dylib target is set, the script assumes ANDROID_NDK env is also set.
+
+PREFIX="module"
+while [[ $# -gt 0 ]]; do
+  token="$1"
+  case $token in
+    --model=*)
+      MODEL=${1#*=}
+      shift
+      ;;
+    --targets=*)
+      TARGETS=${1#*=}
+      shift
+      ;;
+    --prefix=*)
+      PREFIX=${1#*=}
+      shift
+      ;;
+  esac
+done
+
+if [[ -z "${MODEL}" ]]; then
+  echo "Must set --model flag.";
+  exit 1
+fi
+
+export IREE_LLVMAOT_LINKER_PATH="${ANDROID_NDK?}/toolchains/llvm/prebuilt/linux-x86_64/bin/aarch64-linux-android29-clang++ -static-libstdc++ -O3"
+
+IFS=',' read -ra targets_array <<< "$TARGETS"
+for target in "${targets_array[@]}"
+do
+  echo "Compile the module for ${target}..."
+  module_name="${PREFIX}-${target}.vmfb"
+  case "${target}" in
+    "vulkan")
+      build/host/iree/tools/iree-translate \
+        --iree-mlir-to-vm-bytecode-module \
+        --iree-hal-target-backends=vulkan-spirv \
+        ${MODEL} \
+        -o "${module_name}"
+      ;;
+    "vmla")
+      build/host/iree/tools/iree-translate \
+        --iree-mlir-to-vm-bytecode-module \
+        --iree-hal-target-backends=vmla \
+        ${MODEL} \
+        -o "${module_name}"
+      ;;
+    "dylib")
```

Comment:
It seems like we've come up with yet another set of names for this. Let's just use the backend names and keep thing consistent. If we want to make names pithier, we could have unambiguous substring matching, but using a bunch of different aliases for the same thing is super confusing.

Reply:
I'm using driver names here because now they are one to one mapping. I'm not pretty sure the convention, is it DRIVER-XXX? If so, I'm happy to use backend names, and do a simple parsing in another script.

Reply:
Right but this is a compile script so it's weird that it takes driver options. Eventually we should have target triples so that there's a standard way to do this, but for now, let's not muddy things further, so please use the same name as the `--ire-hal-target-backends` argument takes.

Reply:
cool, totally agree that not muddy things further, thanks!